KenticoCloud.Delivery/CodeFirst/CodeFirstModelProvider.cs

```diff
@@ -173,7 +184,89 @@ internal object GetContentItemModel(Type t, JToken item, JToken modularContent,
+                    value = ProcessContentItemsInRichText(modularContent, processedItems, value, modularContentInRichText, currentlyResolvedRichStrings);
+                    currentlyResolvedRichStrings.Remove(currentlyProcessedString);
+                }
+                if (value != null)
+                {
+                    property.SetValue(instance, value);
+                }
+
+            }
+
             return instance;
         }
+
+        private string ProcessContentItemsInRichText(JToken modularContent, Dictionary<string, object> processedItems, string value, JToken modularContentInRichText, HashSet<RichTextContentItem> currentlyResolvedRichStrings)
+        {
+            var usedCodenames = JsonConvert.DeserializeObject<IEnumerable<string>>(modularContentInRichText.ToString());
+            var contentItemsInRichText = new Dictionary<string, object>();
+                
+            foreach (var codenameUsed in usedCodenames)
+            {
+                object contentItem;
+                // This is to reuse content items which were processed already, but not those 
+                // that are calling this resolver as they  may contain unprocessed rich text elements
+                if (processedItems.ContainsKey(codenameUsed) && currentlyResolvedRichStrings.All(x => x.ContentItemCodeName != codenameUsed))  
+                                                                                                                                               
+                {
+                    contentItem = processedItems[codenameUsed];
+                }
+                else
+                {
+                    var modularContentNode = (JObject)modularContent;
+                    var modularContentItemNode =
+                        modularContentNode.Properties()
+                            .FirstOrDefault(p => p.Name == codenameUsed)?.First;
+                    if (modularContentItemNode != null)
+                    {
+                        contentItem = GetContentItemModel(typeof(object), modularContentItemNode, modularContentNode, processedItems, currentlyResolvedRichStrings);
+                        if (!processedItems.ContainsKey(codenameUsed))
+                        {
+                            processedItems.Add(codenameUsed, contentItem);
+                        }
+                    }
+                    else
+                    {
+                        contentItem = new UnretrievedContentItem();     //This means that response from Delivery API didn't contain content of this item 
+                    }
+                }
+                contentItemsInRichText.Add(codenameUsed, contentItem);
+            }
+            value = _client.InlineContentItemsProcessor.Process(value, contentItemsInRichText);
+
+            return value;
+        }
+    
+
+        private string RemoveContentItemsFromRichText(string value)
```

Comment:
It could be removed, however as Martin said, large code blocks are nasty, @kenticomartinh not sure if internal is good practice for testing (making method internal so it can be tested)